parse errors improvements

iliabylich opened the pull request:
+ removed 'Parse Error' prefix
+ included token name into 'unexpected token' error

Also part of https://github.com/sorbet/sorbet/pull/1670

<!-- (optional) Explain your change, focusing on the details of the solution. This is a great place to call out user-visible changes. -->


### Motivation
<!-- Why make this change? Describe the problem, not the solution. This can also be a link to an issue. -->


### Test plan
<!-- If you did not write tests for this change, replace the message below explaining why not. Why we should be confident this change is correct? If you changed the website, please include a screenshot of the proposed changes. -->

See included automated tests.

elliottt commented on parser/Parser.cc:
```diff
@@ -31,24 +31,18 @@ class ErrorToError {
                           translatePos(diag.location().endPos, maxOff));
             if (auto e = gs.beginError(loc, core::errors::Parser::ParserError)) {
-                e.setHeader("Parse {}: {}", level, fmt::format(dclassStrings[(int)diag.error_class()], diag.data()));
+                e.setHeader("{}", fmt::format(dclassStrings[(int)diag.error_class()], diag.data()));
```
I think I might be missing context from the original PR, but why remove the `Parse Error` prefix?

iliabylich replied:
That's the change requested in https://github.com/sorbet/sorbet/pull/1670#discussion_r322410454.

Honestly I like it because my target goal is to rewrite sorbet/parser to match whitequark/parser, and so this way there's no need to do any preprocessing of error messages before backporting tests from whitequark/parser

elliottt replied:
👍 Thanks!

elliottt commented on third_party/parser/cc/grammars/typedruby.ypp:
```diff
@@ -393,14 +395,24 @@ namespace typedruby25 {
 
 int yylex(parser::semantic_type *lval, ruby_parser::typedruby25 &driver) {
 	auto token = driver.lex.advance();
+  driver.last_token = token;
```
Could you indent this with a tab to match the local indentation?

elliottt commented on third_party/parser/include/ruby_parser/driver.hh:
```diff
@@ -3,169 +3,164 @@
-}
+} // namespace ruby_parser
 
 #endif
```
It's pretty difficult to tell what changed in this file given the amount of formatting changes. Was this just re-formatting, or are there semantic changes as well?

iliabylich replied:
only 1 line has changed: `token_t last_token`, the rest is just an auto-formatting

elliottt replied:
Thanks!

iliabylich commented:
@elliottt I've rebased my branch, fixed your comments and updated  https://github.com/sorbet/sorbet/commit/5b610553607e42fb31586b7fc06680fe54d66599 to have a new error message

elliottt commented:
Thank you!

iliabylich commented:
Looks like the build got stuck. @elliottt could you restart it please?

elliottt commented:
I'm not sure of a better way to restart it, but I changed the title to include `[wip]` and then changed it back.
